ygen/enumgen.go

```diff
@@ -127,6 +128,17 @@ func (s *enumSet) enumeratedUnionEntry(e *yang.Entry, compressPaths, noUnderscor
 				return nil, err
 			}
 
+			var enumKind EnumeratedValueType
+			if len(enumNameSake.Type) > 0 {
+				if util.IsYANGBaseType(e.Type) {
```

Comment:
You can likely write this in a more succinct way:


```go
enumKind = DerivedEnumerationType
if len(enumNameSake.Type) > 0 {
  enumKind = DerivedUnionEnumerationType
  if util.IsYANGBaseType(e.Type) {
     enumKind = UnionEnumerationType
  }
}
```

or alternatively:

```go
switch {
  case len(enumNameSake.Type) > 0 && util.IsYANGBaseType(e.Type):
     enumKind = UnionEnumerationType
  case len(enumNameSake.Type) > 0:
     enumKind = DerivedUnionEnumerationType
  default:
     enumKind = DerivedEnumerationType
}
```

the latter seems cleaner to me.

Reply:
Done.